app/controllers/manage_proposal_controller.rb

```diff
@@ -56,6 +56,8 @@ def show
   def publish_proposal
     proposal = JSON.parse(params['proposal_data'])
 
+    add_dates(draft: proposal['draft'])
```

Comment:
An alternate way that reduces duplicate code is something like:
```
proposal_with_dates = CollectionDraftProposal.new(draft: proposal['draft']).add_metadata_dates
proposal = proposal_with_dates.draft
```

Reply:
I didn't do this because the rest of the `manage_proposal_controller` is set up to process the `proposal` variable as a hash instead of a `CollectionDraftProposal` object which causes multiple failures. Good examples of this are the `publish_create_or_update_proposal`, `update_proposal_status_in_dmmt`, and `publish_delete_proposal`. Here are our options: 
1. change the paradigm for how this controller processes proposals - a lot of careful adjustments and potentially some rewriting
2. Use the `CollectionDraftProposal` solely as a means to access the `add_metadata_dates` method and set `proposal['draft'] = proposal_with_dates.draft` essentially just updating this part of the `proposal` hash. 

Thoughts?

Reply:
Had to make some slight changes for this to work